app/src/main/java/nl/tudelft/bbw/controller/KeyReader.java

```diff
@@ -49,7 +49,8 @@ private static void initialize(String filePath) throws IOException {
      * @param path given filepath
      * @return byte array containing the read key
      */
-    public static byte[] readKey(String path) throws IOException {
+    public static byte[] readKey(String path, String password) throws IOException {
+        decryptFile(path, password);
```

Comment:
Because you cannot overwrite a file, while you are reading it and the library does that, if it encrypts a file.